### ISSUE TITLE
Process for releasing new version

### ISSUE THREAD

**Issue by agershun** [agershun/alasql#261]
@mathiasrw If we have problems in package.json, this is means that we need to republish version in npm, right? If it is so, I will technically upgrade AlaSQL to 0.1.9 tongiht/

**Comment by mathiasrw** [comment]
Yes! 

Sounds awesome

**Comment by agershun** [comment]
Done.

**Comment by agershun** [comment]
@mathiasrw May I ask you to check this page https://github.com/agershun/alasql/wiki/How-to-prepare-new-release

**Comment by mathiasrw** [comment]
Sure. Ill cant right now but get back to it as soon as possible. Tonight or tomorrow...

**Comment by mathiasrw** [comment]
Some inputs on how to make it better:

----

> // Change version in README.md

With the setup we are having with the readme on the wiki its probably better to write something along with

- Change version in wiki README
- Copy wiki README into README.md

----

> // Change version in package.json
> // Change version in .bower.json

ok

----

> // Check CHANGELOG.md

Probably best to write whats to be done to CHANGELOG.md

----

> // Change version in src/05start.js
> // Change version in src/10alasql.js

Aw... would be smooth if we can have the version in only one alasql js file

----

> // Change version in meteor/10alasql.js

ok

----

> git add --all
> git commit -m 'Release 0.1.10'

Hmmm... normally I would expect everything to be comitted already - but ok - it might be handy. 

----

> git flow release start 0.1.10
> // resolve all merge conflicts

I must say I wonder what conflicts occures here for you - we are making a new branch and there should be no merging. Could you elaborate on the issues you get here?

----

> git flow release finish 0.1.10

I have a feeling we are missing the point of the release step. The relese step is where we make sure everything is as its ment to be. By that I would expect this list to include some of the following before finishing the release. 

- Compile alasql som src
- Run main tests (not SQLLOCIG nor NIST) 
- Verify all tests are OK 
- If all tests not OK - fix it (Remember that then a release is finished it gets merged into both master and development - so any changes we are doing at the release will get into the normal flow when we "finish" it.)
- (Could also be here we do all the opdates mentioned prior to "the git add")


----



> git checkout develop
> git push --tags
> git checkout develop

I dont quite get the double checkout. The following should be what we need (but please test):

    git push --all --tags


----

The following commands must be done from Master or from release just before we finish. In theory we dont know if we have introduced any changes to development while we had the release open. 

> npm publish
> cd meteor
> meteor publish

A reason to put it in as last part of the release step is that if we find that something is wrong with the config files we can still edit while keeping a clean repo history. 


-----


> // Move description from CHANGELOG.md to github.com/agershun/alasql
> // Create new header for CHANGELOG.md

I dont get this. If CHANGELOG is in the repo - it should already be in master now. All changes should be before the release is "finished" So suggest we move it up into the release stage.

**Comment by agershun** [comment]
:+1:  Mathias thank you for detail analysis! It was definitely not perfect :)  I will update the procedure and we can discuss it again.

**Comment by mathiasrw** [comment]
Sounds good. 

But specially the merging conflict part is not good. It must be easy and smooth :)

**Comment by mathiasrw** [comment]
We really need to get good at this - also when its just a little change... 

In the moment:

- Latest github release is 0.1.8 https://github.com/agershun/alasql/releases
- README.md on master says "`Version: 0.1.9 "Torino" Updated: May 31, 2015,`"
- npm version number states `0.1.10`

Lets write a good and easy to follow step by step check list to make it all nice and smooth...

**Comment by agershun** [comment]
I added the missed entries for a now. But you are right, we need more formal procedures on this prorcess. Will work on it.

**Comment by agershun** [comment]
This is also important, because the updating tags is the trigger for jsDeliver ( see https://github.com/jsdelivr/jsdelivr/pull/5215)

**Comment by mathiasrw** [comment]
Argh - so many places...

In the moment, if I go into https://cdn.rawgit.com/agershun/alasql/0.1.10/dist/alasql.js

the URL indicates 0.1.10 + but the first lines contains the following:

```
/*! AlaSQL v0.1.9 (c) 2014-2015 Andrey Gershun | alasql.org/license */
/*
@module alasql
@version 0.1.8

AlaSQL - JavaScript SQL database
...
```

**Comment by agershun** [comment]
:) Plus we have a problem with Meteor (I have made a mistake and now AlaSQL version is +10 to the original version (at least for 0.1.xx)).

We can generate the nummber by simple preporcessing at ```gulp ``` assembling time with simple templater (e.g.warer-tmplt). For example put 

BTW. I received letter ftom author of ```gauis``` preprocessor:
> 
I had a look at the repository https://github.com/agershun/alasql for the alasql package that you are maintaining. I see that you are using gulp.js. I have built a GitHub markdown preprocessor Gitdown (https://github.com/gajus/gitdown). It adds additional functionality (e.g. generating table of contents, including documents, using variables, etc.) to the GitHub markdown. It integrates easily with Gulp and it will help you to maintain the documentation.
